config/routes.rb

```diff
@@ -1,22 +1,14 @@
 Webapp::Application.routes.draw do
 
-  resources :home do
+  resources :assignments, only: [] do
     collection do
-      get :welcome
-    end
-  end
-
-  resources :assignments do
-    collection do
-      get :index
```

Comment:
Interesting find!  As far as I can tell, this link doesn't work anyway.  There isn't a view defined for that action. Does it work for you?

---

Aside, this code isn't using Rails properly:

```ruby
resources :assignments do
  collection do
    get :index
    #...
  end
end
```

This is defining a redundant `index` action.  `resources :assignments` will automatically set up 7 routes - `index`, `show`, `new`, `create`, `edit`, `update`, and `destroy`.

The DSL methods - `collection` and `member` - are only useful to add non-RESTful routes to a resource.

The routing above creates two indicies:

```
‡ be rake routes | grep assignments
GET      /assignments(.:format)                          assignments#index
GET      /assignments/knight(.:format)                   assignments#knight
GET      /assignments(.:format)                          assignments#index
POST     /assignments(.:format)                          assignments#create
#...
```

See http://guides.rubyonrails.org/routing.html#adding-more-restful-actions